File: Questao 1/Pilha.cs
namespace Codes;
public class Pilha
{
    public int Top { get; set; }
    public Object[] Vetor { get; set; }

    public Pilha(int tamanho)
    {
        Top = -1;
        Vetor = new Object[tamanho];
    }
    public int PosTop()
    {
        return Top;
    }
    public Object ShowTop()
    {
        if (IsEmpty()) return null;
        return Vetor[Top];
    }
    public bool IsEmpty()
    {
        return Top == -1;
    }
    public bool isFull()
    {
        return Top == Vetor.Length - 1;
    }
    public void Push(Object obj)
    {
        if (!isFull())
        {
            Vetor[++Top] = obj;
        }
        else
        {
            System.Console.WriteLine("Pilha cheia!");
        }
    }
    public Object Pop()
    {
        if (!IsEmpty())
        {
            var obj = Vetor[Top];
            Vetor[Top] = null;
            Top--;
            return obj;
        }
        return null;
    }
    public bool Search(Object obj)
    {
        if (!IsEmpty())
        {
            for (int i = Top; i >= 0; i--)
            {
                if (Vetor[i] == obj)
                    return true;
            }
        }
        return false;
    }
    public void Show()
    {
        if (!IsEmpty())
        {
            for (int i = Top; i >= 0; i--)
            {
                System.Console.WriteLine(Vetor[i]);
            }
            System.Console.WriteLine();
        }
        else
        {
            System.Console.WriteLine("Pilha Vazia!");
        }
    }
}

File: AC3/Questao 14/Pilha.java
//questão14 parteII

public class Pilha{
    
    private int top;
    private Object stack[];
    public int length;

    public Pilha(int length){
        stack = new Object[length];
        top = -1;
    }

    public void push(String obj){
        if(isFull()){
            System.out.println("está cheia");
        }
        else{
                stack[++top] = obj;
        }
    }

    public Object pop(){
        if(isEmpty()){
            return null;
        }
        else{
            Object x = stack[top];
            stack[top] = null;
            top--;
            return x;
        }
    }
    
    public int top(){
        return this.top;
    }


    public boolean isEmpty(){
        if(this.top < 0){
        return true;
    }
        return false;
    }
   


    public boolean isFull(){
        if(this.top == stack.length-1){
            return true;
        }
        return false;
    }


    public void show(){
        if(isEmpty()){
            System.out.println("pilha vazia");
        }else{
        for(int i = this.top; i>=0; i--){
            System.out.println(stack[i]);
        }
    }
    }

    public Object topShow(){
        if(isEmpty()){
            return null;
        }
        return this.stack[length];
    }


 }
    






